# EJEMPLO 8.9: ENERGIA DEL ESTADO FUNDAMENTAL EN UNA CAJA CUADRADA

Vamos a calcular la energia del estado fundamental de un electron en un potencial de una caja cuadrada con paredes altas indefinidas separas una distancia $L$ igual al radio de Bohr $a_{0}=5.292 \times 10^{-11} m$. Aqui hay un prgrama para hacer el calculo usando el metodo de la secante:

In [8]:
from numpy import array, arange

# Constantes
m = 9.1094e-31      # Masa del electron
hbar = 1.0546e-34   # Constante de Planck sobre 2*pi
e = 1.6022e-19      # Carga del electron
L= 5.2918e-11      # Radio de Bohr
N = 1000
h = L/N

# Funcion Potencial
def V(x):
    return 0.0

def f(r,x,E):
    psi = r[0]
    phi = r[1]
    fpsi = phi
    fphi = (2*m/hbar**2)*(V(x)-E)*psi
    return array([fpsi,fphi],float)

# Calcular la longitud de onda para una energia particular
def solve(E):
    psi = 0.0
    phi = 1.0
    r = array([psi,phi],float)
    
    for x in arange(0,L,h):
        k1 = h*f(r,x,E)
        k2 = h*f(r+0.5*k1,x+0.5*h,E)
        k3 = h*f(r+0.5*k2,x+0.5*h,E)
        k4 = h*f(r+k3,x+h,E)
        r += (k1+2*k2+2*k3+k4)/6
        
    return r[0]

# Programa principal para encontrar la energia usando el metodo de la secante
E1 = 0.0
E2 = e
psi2 = solve(E1)

target = e/1000
while abs(E1-E2)>target:
    psi1,psi2 = psi2,solve(E2)
    E1,E2 = E2,E2-psi2*(E2-E1)/(psi2-psi1)
    
print("E =",E2/e,"eV")

E = 134.28637169369105 eV


Si corremos el programa, este imprime la energía del estado fundamental por lo tanto:

$E = 134.286371694 eV$

el cual es en efecto la respuesta correcta. Note que la funcion $V(x)$ no hace nada en estre programa, desde que el potencial en cualquier lugar de la caja sea cero, no juega ningun papel en el calculo. Sin embargo, al incluir la función en el programa hacemos mas facil resolver otro, menos potencial trivial en los problemas de cajas. Por ejemplo, supongamos que potencial dentro de la caja no es cero pero varia como 

\begin{equation}
    V(x)=V_{0}\frac{x}{L}\left ( \frac{x}{L}-1 \right ),
\end{equation}

donde $V_{0}=100eV$. Es una cuestión simple para resolver la energia del estado fundamental de este problema tambian. Solo debemos cambiar la función $V(x)$, así:

In [9]:
from numpy import array, arange

# Constantes
m = 9.1094e-31      # Masa del electron
hbar = 1.0546e-34   # Constante de Planck sobre 2*pi
e = 1.6022e-19      # Carga del electron
L= 5.2918e-11      # Radio de Bohr
N = 1000
h = L/N
V0 = 100*e

# Funcion Potencial
def V(x):
    return V0*(x/L)*(x/L-1)

def f(r,x,E):
    psi = r[0]
    phi = r[1]
    fpsi = phi
    fphi = (2*m/hbar**2)*(V(x)-E)*psi
    return array([fpsi,fphi],float)

# Calcular la longitud de onda para una energia particular
def solve(E):
    psi = 0.0
    phi = 1.0
    r = array([psi,phi],float)
    
    for x in arange(0,L,h):
        k1 = h*f(r,x,E)
        k2 = h*f(r+0.5*k1,x+0.5*h,E)
        k3 = h*f(r+0.5*k2,x+0.5*h,E)
        k4 = h*f(r+k3,x+h,E)
        r += (k1+2*k2+2*k3+k4)/6
        
    return r[0]

# Programa principal para encontrar la energia usando el metodo de la secante
E1 = 0.0
E2 = e
psi2 = solve(E1)

target = e/1000
while abs(E1-E2)>target:
    psi1,psi2 = psi2,solve(E2)
    E1,E2 = E2,E2-psi2*(E2-E1)/(psi2-psi1)
    
print("E =",E2/e,"eV")

E = 112.54010720768352 eV


Entonces corremos el programa de nuevo y esta vez encontramos que la energia del estado fundamental es

$E = 112.540107208 eV$

Aunque el problema original de la caja cuadrada es facil de resolver analiticamente, esta segunda versión del programa con un potencial variable seria mucho mas dificil, and yet a solution is achueved easily using the computer.